js_modules/dagit/packages/core/src/launchpad/LaunchpadAllowedRoot.tsx

```diff
@@ -0,0 +1,212 @@
+import {gql, useQuery} from '@apollo/client';
+import * as React from 'react';
+
+import {IExecutionSession} from '../app/ExecutionSessionStorage';
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
+import {useTrackPageView} from '../app/analytics';
+import {explorerPathFromString, useStripSnapshotFromPath} from '../pipelines/PipelinePathUtils';
+import {useJobTitle} from '../pipelines/useJobTitle';
+import {useRepository, isThisThingAJob} from '../workspace/WorkspaceContext';
+import {RepoAddress} from '../workspace/types';
+
+import {
+  CONFIG_EDITOR_GENERATOR_PARTITION_SETS_FRAGMENT,
+  CONFIG_EDITOR_GENERATOR_PIPELINE_FRAGMENT,
+} from './ConfigEditorConfigPicker';
+import {LaunchpadSessionError} from './LaunchpadSessionError';
+import {LaunchpadSessionLoading} from './LaunchpadSessionLoading';
+import {LaunchpadTransientSessionContainer} from './LaunchpadTransientSessionContainer';
+import {LaunchpadType} from './types';
+import {LaunchpadRootQuery, LaunchpadRootQueryVariables} from './types/LaunchpadRoot.types';
+
+const LaunchpadStoredSessionsContainer = React.lazy(
+  () => import('./LaunchpadStoredSessionsContainer'),
+);
+
+interface Props {
+  launchpadType: LaunchpadType;
+  pipelinePath: string;
+  repoAddress: RepoAddress;
+  sessionPresets?: Partial<IExecutionSession>;
+}
+
+export const LaunchpadAllowedRoot = (props: Props) => {
```

Comment:
Split this out into its own component to avoid circular dependencies that I hit when trying to mock it for the test.